# origin: marcelo (BitKeeper)
# cset: 1.1559 (2.4) key=41e2c5fb3htiRRycYu5I4skGWXcv5g
# inclusion: upstream
# descrition: Alan Cox: Fix moxa serial bound checking issue (from 2.6.10-ac)
# revision date: Thu, 13 Jan 2005 15:16:21 +0900
#
# S rset: ChangeSet|1.1558..1.1559
# I rset: drivers/char/moxa.c|1.8..1.9
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/10 16:14:19-02:00 marcelo@logos.cnet 
#   Alan Cox: Fix moxa serial bound checking issue (from 2.6.10-ac)
# 
# drivers/char/moxa.c
#   2005/01/10 16:11:04-02:00 marcelo@logos.cnet +2 -0
#   Alan Cox: Fix moxa serial bound checking issue
# 
#
===== drivers/char/moxa.c 1.8 vs 1.9 =====
--- 1.8/drivers/char/moxa.c	2004-12-17 00:14:38 +09:00
+++ 1.9/drivers/char/moxa.c	2005-01-11 03:11:04 +09:00
@@ -905,6 +905,8 @@
 	case TIOCSSERIAL:
 		return (moxa_set_serial_info(ch, (struct serial_struct *) arg));
 	default:
+		if(!capable(CAP_SYS_RAWIO))
+			return -EPERM;
 		retval = MoxaDriverIoctl(cmd, arg, port);
 	}
 	return (retval);
